### ISSUE TITLE
Detailed events such as when the image is being pulled, and when the container is started are missing

### ISSUE THREAD

**Issue by xinyanmsft** [virtual-kubelet/virtual-kubelet#472]
Provider (e.g. ACI, AWS Fargate, Hyper)  ACI
Version (e.g. 0.1, 0.2-beta)  Current
K8s Master Info (e.g. AKS, ACS, Bare Metal, EKS)   AKS
Install Method (e.g. Helm Chart, ) Helm (via AKS)

Right now, running ‘kubectl events’ doesn't give me a lot of useful information, as compare to regular Kubernetes nodes. Detailed events such as when the image is being pulled, and when the container is started are not there. These are helpful in general for developers to understand what's going on and troubleshooting.

**Comment by cpuguy83** [comment]
I don't think this is something we can necessarily support in the VK core and seems like a provider concern, though certainly the existing providers are not reporting this information.

**Comment by xinyanmsft** [comment]
@cpuguy83 @rbitia  please advice where I can move this issue to. I do care about AKS / ACI. Thanks!

**Comment by pires** [comment]
I'd say this is specific to a provider. For instance, for ACI it may make sense to report such events, while for other providers not. The provider may not even have access to such info, i.e. _pulled an image_.

Also, this relates to #443 where @Smithx10 has expressed the desire to record events from providers too.

**Comment by rbitia** [comment]
@xinyanmsft This is on our schedule for next semester. Thanks for the issue :)
